tfx/components/transform/executor.py

```diff
@@ -331,8 +331,7 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
       splits_config.analyze.append('train')
       splits_config.transform.extend(
           artifact_utils.decode_split_names(
-              artifact_utils.get_single_instance(
-                  input_dict[EXAMPLES_KEY]).split_names))
+              input_dict[EXAMPLES_KEY][0].split_names))
```

Comment:
let's use the union set of all splits for transform (or verify all input has the same splits name)?

Reply:
i think the latter option will probably be more well defined (and be more clear to users in case of an err), will fix that